nouvelles-perspectives-des-sciences-sociales.csl

```diff
@@ -0,0 +1,606 @@
+        </names>
+      </else>
+    </choose>
+  </macro>
+  <macro name="translator">
+    <names variable="translator">
+      <label form="verb" suffix=" "/>
+      <name and="text" delimiter-precedes-last="never" initialize="false" sort-separator=" "/>
+    </names>
+  </macro>
+  <macro name="editor">
+    <group>
+      <names variable="editor" delimiter=", ">
+        <name and="text" delimiter-precedes-last="never" initialize="false" name-as-sort-order="all">
+          <name-part name="family"/>
+        </name>
+        <label form="short" prefix=" (" suffix=")"/>
+      </names>
+    </group>
+  </macro>
+  <macro name="editor-notes">
+    <group>
+      <names variable="editor" delimiter=", ">
+        <name and="text" delimiter-precedes-last="never" initialize="false">
+          <name-part name="family"/>
+        </name>
+        <label form="short" prefix=" (" suffix=")"/>
+      </names>
+    </group>
+  </macro>
+  <macro name="container-contributors">
+    <choose>
+      <if variable="editor container-author" match="any">
+        <text term="in" text-case="lowercase"/>
+        <choose>
+          <if variable="container-author">
+            <names variable="container-author">
+              <name and="text" delimiter-precedes-last="never" initialize="false">
+                <name-part name="family"/>
+              </name>
+            </names>
+          </if>
+          <else-if variable="editor">
+            <group>
+              <names variable="editor" delimiter=" ">
+                <name and="text" delimiter-precedes-last="never" initialize="false">
+                  <name-part name="family"/>
+                </name>
+                <label form="short" prefix=" (" suffix=")"/>
+              </names>
+            </group>
+          </else-if>
+        </choose>
+      </if>
+    </choose>
+  </macro>
+  <macro name="title">
+    <choose>
+      <if type="thesis" match="any" variable="container-title">
+        <text variable="title" text-case="capitalize-first" quotes="true" font-style="normal"/>
+      </if>
+      <else>
+        <text variable="title" text-case="capitalize-first" quotes="false" font-style="italic"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="online">
+    <choose>
+      <if variable="URL DOI" match="any">
+        <group delimiter=", ">
+          <text term="online" text-case="lowercase"/>
+          <choose>
+            <if match="any" variable="URL">
+              <text variable="URL" prefix="&lt;" suffix="&gt;"/>
+            </if>
+            <else>
+              <text variable="DOI" prefix="&lt;doi: " suffix="&gt;"/>
+            </else>
+          </choose>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="publisher">
+    <choose>
+      <if type="manuscript graphic legislation legal_case bill article-journal article-magazine article-newspaper review-book interview speech treaty dataset" match="none">
+        <group delimiter=", ">
```

Comment:
this group doesn't do anything

Reply:
We are sorry. We will make all the suggested corrections

Reply:
The comma will not appear ? Do we have to cancel it ?